core/unsigneddata_test.go

```diff
@@ -24,30 +24,41 @@ import (
 	"github.com/obolnetwork/charon/testutil"
 )
 
-func TestCloneVersionedBeaconBlock(t *testing.T) {
-	block := testutil.RandomCoreVersionBeaconBlock(t)
-	slot1, err := block.Slot()
-	require.NoError(t, err)
-
-	clone, err := block.Clone()
-	require.NoError(t, err)
-	block2 := clone.(core.VersionedBeaconBlock)
-	slot2, err := block2.Slot()
-	require.NoError(t, err)
-
-	require.Equal(t, slot1, slot2)
-}
+func TestUnsignedDataClone(t *testing.T) {
+	tests := []struct {
+		name string
+		data core.UnsignedData
+	}{
+		{
+			name: "attestation data",
+			data: testutil.RandomCoreAttestationData(t),
+		},
+		{
+			name: "versioned beacon block",
+			data: testutil.RandomCoreVersionBeaconBlock(t),
+		},
+		{
+			name: "versioned blinded beacon block",
+			data: testutil.RandomCoreVersionBlindedBeaconBlock(t),
+		},
+		{
+			name: "aggregated attestation",
+			data: core.NewAggregatedAttestation(testutil.RandomAttestation()),
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			clone, err := test.data.Clone()
+			require.NoError(t, err)
 
-func TestCloneVersionedBlindedBeaconBlock(t *testing.T) {
-	block := testutil.RandomCoreVersionBlindedBeaconBlock(t)
-	slot1, err := block.Slot()
-	require.NoError(t, err)
+			b1, err := test.data.MarshalJSON()
+			require.NoError(t, err)
 
-	clone, err := block.Clone()
-	require.NoError(t, err)
-	block2 := clone.(core.VersionedBlindedBeaconBlock)
-	slot2, err := block2.Slot()
-	require.NoError(t, err)
+			b2, err := clone.MarshalJSON()
+			require.NoError(t, err)
 
-	require.Equal(t, slot1, slot2)
+			require.Equal(t, b1, b2)
```

Comment:
Let's also add another line of test for `UnmarshalJSON()`. Like, for VersionedBeaconBlock:
```go
	x := new(core.VersionedBeaconBlock)
	require.NoError(t, x.UnmarshalJSON(b1))
	require.Equal(t, *x, test.data)
```

Reply:
i don't think we need this since Clone() method already takes care of this

Reply:
Btw, Clone() also uses the `MarshalJSON()` method, so actually, we don't need to test the marshal/unmarshal methods explicitly.

Reply:
sure